engine/expr.h

```diff
@@ -156,6 +156,12 @@ namespace MathDAG
     r->arguments[0].push_back(x);
     return Expr(x.cache,r);
   }
+  
+  //inline Expr fact(const Expr& x) {
+  //  shared_ptr<OperationDAGBase> r(x.newNode(OperationType::fact));
+  //  r->arguments[0].push_back(x);
+  //  return Expr(x.cache,r);
+  //}  
```

Comment:
This code is fine. We need it, in case someone attaches the derivative operator to an expression containing a factorial.

Reply:
Okay, I have put it back in.